### PR TITLE
feat(itb): scrap pdf URL if not available

### PR DESCRIPTION
**Pull request by irynadosiak-c-change-lab**
asana: [Scrap ITB pdf URL if not available in provider website](https://app.asana.com/0/698742722983597/1207259709763382/f)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -28,6 +28,7 @@ urllib3 = { version = ">=1.26.17,<2.0.0" }
 requests = { version = ">=2.1.0,<3.0.0" }
 openepd = { version = ">=3.1.0" }
 pytz = { version = ">=2023.3" }
+beautifulsoup4 = { version = ">=4.10.0" }
```
Comment (corvis) [inline pyproject.toml:31]: Why do we need beatutiful soap if we already have and use mechanical soup?
See `cqd_etl.ingestion.common.scrapping.impl.msoap.BaseMechanicalSoapSpider` and it's child classes for reference.

**Review by corvis — CHANGES_REQUESTED** [review]

**Comment by corvis** [comment]
Ok, I didn't realized it is in ilcdlib. I don't want this logic to be a part of ILCDLIB, it should  in in the airflow.

**Comment by irynadosiak-c-change-lab** [comment]
> Ok, I didn't realized it is in ilcdlib. I don't want this logic to be a part of ILCDLIB, it should in in the airflow.

What if we need to use ILCDLIB in contexts beyond Airflow? Should we also implement the logic there?

**Comment by corvis** [comment]
This is a public library and i don't want it to do anything beyond ILCD related things.

**Comment by irynadosiak-c-change-lab** [comment]
> This is a public library and i don't want it to do anything beyond ILCD related things.

Ok, will update

**Pull request closed without merging**